### DISCUSSION TITLE
More fun with grouptopics in shutter mode

### DISCUSSION THREAD

**Discussion by Bascht74** [arendst/Tasmota#10200] [Ideas]
@stefanbode Hi, I am getting more into the configuration of the shutters now and I am right now at the point where I want to configure the buttons of the shutters.

In the documentation I found:
`ShutterButton<x> <button> <p1> <p2> <p3> <ph> <m1> <m2> <m3> <mh> <mi>`
where e.G. `<m1> 1: enable single press position MQTT broadcast, 0/-: disable`

`Global steering of all your shutters at home is supported by additional MQTT broadcast. By any button action a corresponding MQTT command can be initiated to the <grouptopic> of the device. For single press this can be enabled by <m1> equal to 1, disabling is indicated by -. Double to hold MQTT configurations are given by <m2> to <mh>, correspondingly. When <mi> is equal to - only cmnd/<grouptopic>/Shutterposition<x> <p1..h> is fired. When <mi> is equal to 1, <x>=1..4 is used to control any shutter number of a tasmota device having same <grouptopic>.`

I have four group topic configured: 
`{"GroupTopic1":"rollos_alle","GroupTopic2":"rollos_erdgeschoss","GroupTopic3":"rollos_wohnzimmer","GroupTopic4":"tasmotas"}`

I would now only have the the MQTT broadcast of the GroupTopic3 triggered. 
I don't know how I can configure this. 

Maybe it would be a good idea that e.g.
`<m1> 1-4: enable single press position MQTT broadcast for grouptopic 1-4, 0/-: disable`

example: 
ShutterButton<x> <button> 0 50 24 0 0 0 0 3 3
only triggers the third grouptopic broadcast.

Thanks again for that great integration, 

Sebastian

**Comment by stefanbode** [comment]
The button-press methods were added through another user and util know I tried to avoid to understand all that fancy and tricky code stuff there. The proposed solution does not look very complicated. Anyhow, I need to check on which group-topic the post occurs currently when set m1.

**Reply by Bascht74** [reply]
> Anyhow, I need to check on which group-topic the post occurs currently when set m1.
Great! Maybe i doesn't work as supposed because there are now 4 grouptopics instead of one...

**Comment by stefanbode** [comment]
I checked a bit about how the m1 is used and where the values are stored. Short message: The array to store data is full. No more space left. I have 32bits and all used. I would need additional 8bits (0=off, 1..4) (3bits) x 4 grouptopics = 12 bits and I have currently 4bits (0...1). Adding this feature would require a change in the global storage array and worst case a break to old versions..... further investigations required

**Reply by Bascht74** [reply]
OK, but do you know which `grouptopic` is used? e.G. the first (`grouptopic1`) or all grouptopics? (I will check when I am back at home). 
Maybe it would be enough to add this information to the documentation. 
This way, you can at least store the grouptopic you want in the number that is taken by m1...
Seb

**Reply by stefanbode** [reply]
it is for sure the first grouptopic that is used. topic 2..4 where "invented" after the shutterbutton where coded. Thats the reason it is not in. Anyhow I have an idea.

**Reply by Bascht74** [reply]
OK & and Merry Chrismas!

I am already quite curious about your idea and the patch for the other topic... :-)
